spyder_line_profiler/spyder/widgets.py

```diff
@@ -138,13 +153,13 @@ def __init__(self, parent):
         self.filecombo = PythonModulesComboBox(self)
 
         self.start_button = create_toolbutton(
```

Comment:
The actions creation should be in the `setup` method. I think a good example of the structure this should follow is the Profiler plugin: https://github.com/spyder-ide/spyder/blob/master/spyder/plugins/profiler/widgets/main_widget.py

Also, instead of creating the actions as tool buttons, lets create them as actual actions to add them shortcuts so for example the `start_button` should become `start_action` and be create with something like:

```python
self.start_action = self.create_action(
            LineProfilerWidgetActions.Run,
            text=_("Profile by line"),
            tip=_("Run line profiler"),
            icon=self.create_icon('run'),
            triggered=self.analyze,
        )
```

Where the `LineProfilerWidgetActions` works as a holder for constants to identify the actions and then all the actions create can be added to the default main toolbar that the new API provides with something like the for as is used in the Profile plugin:

```python
 toolbar = self.get_main_toolbar()
        for item in [self.filecombo, browse_action, self.start_action, self.stop_action]:
            self.add_item_to_toolbar(
                item,
                toolbar=toolbar,
                section=LineProfilerWidgetMainToolbarSections.Main,
            )

```

Where `LineProfilerWidgetMainToolbarSections` defines constants for the sections available for the main toolbar.

Also, for the icons please use the `self.create_icon` method (in that way the plugin can use the new set of icons that the rest of Spyder is currently using). I think the names of the icons to use are for the main toolbar 'fileopen', 'start', 'stop', and for the secondary toolbar `collapse`, `expand` and 'log'.

In that way the layout of the plugin will be more similar to the rest of the migrated plugins